Fix to `pypeit_compare_sky` to render script functional

tbowers7 opened the pull request:
Bring `pypeit_compare_sky` script up to speed with `spec1d` data model version 1.1.4

      * Specify the FITS bintable is in extension 1
      * Update tags for table fields to match current specs
      * Specify in help that user submits a `spec1d` file
      * Update doc to reflect updated help

	modified:   pypeit/scripts/compare_sky.py
	modified:   doc/help/pypeit_compare_sky.rst
---

Unit test discussion below.

tbowers7 committed:
Fix to pypeit_compare_sky to render script functional

Bring script up to speed with spec1d data model version 1.1.4

      * Specify the FITS bintable is in extension 1
      * Update tags for table fields to match current specs
      * Specify in help that user submits a spec1d file
      * Update doc to reflect updated help

	modified:   pypeit/scripts/compare_sky.py
	modified:   doc/help/pypeit_compare_sky.rst

jhennawi reviewed (commented):
Can you work with @badpandabear to add a test to this script to the dev suite to avoid it becoming not functional again?

tbowers7 committed:
Merge branch 'develop' of https://github.com/pypeit/PypeIt into fix_script_compare_sky

tbowers7 committed:
Add test for `pypeit_compare_sky` script

Add a test for this script, which had stopped working because of a change
in the `spec1d` data model.

For the `compare_sky` script itself, add an argument `--test` to prevent
matplotlib from showing the plot, as plt.show() pauses execution until the
plot window cloded.

	modified:   pypeit/scripts/compare_sky.py
	modified:   pypeit/tests/test_scripts.py

tbowers7 commented:
Added test for `pypeit_compare_sky`.  @badpandabear, please look over the new test to ensure it does what is needed.

tbowers7 committed:
Doc update to reflect new argument to `pypeit_compare_sky`

	modified:   doc/help/pypeit_compare_sky.rst

tbowers7 commented:
```(pypeit) tbowers@Zephyr:~/d1/codes/PypeIt$ pytest -v pypeit/tests/test_scripts.py
============================= test session starts ==============================
collected 15 items

pypeit/tests/test_scripts.py::test_quicklook PASSED                      [  6%]
pypeit/tests/test_scripts.py::test_trace_edges PASSED                    [ 13%]
pypeit/tests/test_scripts.py::test_trace_add_rm PASSED                   [ 20%]
pypeit/tests/test_scripts.py::test_show_1dspec PASSED                    [ 26%]
pypeit/tests/test_scripts.py::test_show_2dspec PASSED                    [ 33%]
pypeit/tests/test_scripts.py::test_chk_edges PASSED                      [ 40%]
pypeit/tests/test_scripts.py::test_view_fits PASSED                      [ 46%]
pypeit/tests/test_scripts.py::test_chk_flat PASSED                       [ 53%]
pypeit/tests/test_scripts.py::test_chk_wavecalib PASSED                  [ 60%]
pypeit/tests/test_scripts.py::test_coadd1d_1 PASSED                      [ 66%]
pypeit/tests/test_scripts.py::test_coadd1d_2 PASSED                      [ 73%]
pypeit/tests/test_scripts.py::test_identify PASSED                       [ 80%]
pypeit/tests/test_scripts.py::test_obslog PASSED                         [ 86%]
pypeit/tests/test_scripts.py::test_compare_sky PASSED                    [ 93%]
pypeit/tests/test_scripts.py::test_collate_1d PASSED                     [100%]

================= 15 passed, 546 warnings in 150.68s (0:02:30) =================

tbowers7 commented:
Run of the full test suite has 2 failures when run against Cooked v1.4.5

The `test_setup_vlt_fors2` failure is the same one noted in PR #1216, but the `test_run_on_bad_headers` failure is new.

```
=========================== short test summary info ============================
FAILED pypeit/tests/test_pypeitsetup.py::test_run_on_bad_headers - assert 229...
FAILED pypeit/tests/test_setups.py::test_setup_vlt_fors2 - TypeError: NoneTyp...
========== 2 failed, 287 passed, 4500 warnings in 1543.02s (0:25:43) ===========


=================================== FAILURES ===================================
___________________________ test_run_on_bad_headers ____________________________

    @dev_suite_required
    def test_run_on_bad_headers():
        files = get_lrisr_files()
        # Init
        setupc = pypeitsetup.PypeItSetup(files,
                                         spectrograph_name='keck_lris_red',
                                         path=data_path(''))
        # Run
        par, spectrograph, fitstbl = setupc.run(setup_only=True,
                                                sort_dir=data_path(''))
        # Test
        assert par is None
>       assert setupc.fitstbl['ra'][-1] is None
E       assert 229.99999999999997 is None

/home/tbowers/d1/codes/PypeIt/pypeit/tests/test_pypeitsetup.py:166: AssertionError

_____________________________ test_setup_vlt_fors2 _____________________________

    @dev_suite_required
    def test_setup_vlt_fors2():
        droot = os.path.join(os.environ['PYPEIT_DEV'], 'RAW_DATA/vlt_fors2/300I/')
        droot += '/FORS2'
        pargs = Setup.parse_args(['-r', droot, '-s', 'vlt_fors2'])
        Setup.main(pargs)
...
        # Now chk calib
        pargs = ChkForCalibs.parse_args([droot, '-s', 'vlt_fors2'])
>       answers, ps = ChkForCalibs.main(pargs)

/home/tbowers/d1/codes/PypeIt/pypeit/tests/test_setups.py:536:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = Namespace(root='/home/tbowers/d1/codes/PypeIt-development-suite/RAW_DATA/vlt_fors2/300I//FORS2', spectrograph='vlt_fors2', extension='.fits', save_setups=False)

    @staticmethod
    def main(args):
...
        # Unique configurations
        uniq_cfg = ps.fitstbl.unique_configurations(copy=True)

        # Setup the table. Need to use object type for strings so that
        # they're not truncated.
        answers = table.Table()
        answers['setups'] = list(uniq_cfg.keys())
        # Add the configuration columns
        for setup, setup_dict in uniq_cfg.items():
            for key, value in setup_dict.items():
                answers[key] = np.empty(len(answers), dtype=object) if isinstance(value, str) \
>                                   else type(value)(0)
E               TypeError: NoneType takes no arguments

/home/tbowers/d1/codes/PypeIt/pypeit/scripts/chk_for_calibs.py:90: TypeError

tbowers7 committed:
Merge branch 'develop' of https://github.com/pypeit/PypeIt into fix_script_compare_sky

jhennawi commented:
The bad headers failure occurs on develop, but I don't think the vlt_fors2 one does. I would say switch to develop and the develop branch of the dev suite and confirm that you get the same two failures. If not there is something in this PR breaking vlt_fors2 which I doubt.

tbowers7 commented:
@jhennawi When running the test suite on `develop` against the `develop` branch of the dev suite and Cooked v1.4.5, I get the same two errors:

```
=========================== short test summary info ============================
FAILED pypeit/tests/test_pypeitsetup.py::test_run_on_bad_headers - assert 229...
FAILED pypeit/tests/test_setups.py::test_setup_vlt_fors2 - TypeError: NoneTyp...
========== 2 failed, 286 passed, 4500 warnings in 1564.31s (0:26:04) ===========

profxj reviewed (approved):
Looks great, thx

profxj commented:
I'll fix the two errors above next week.

tbowers7 committed:
Merge branch 'develop' of https://github.com/pypeit/PypeIt into fix_script_compare_sky

badpandabear commented on pypeit/tests/test_scripts.py:
```diff
@@ -327,6 +328,17 @@ def test_obslog():
+    # Running in `test` mode
+    pargs = scripts.compare_sky.CompareSky.parse_args([spec_file, sky_file, '--test'])
+    scripts.compare_sky.CompareSky.main(pargs)
+
```
This test should also run the script with --optimal so that it can catch changes to those keywords as well changes as to the boxcar keywords.

tbowers7 replied:
Added.

badpandabear reviewed (changes requested):
I think adding a unit test here instead of a dev suite test was the right way to go as this is fairly quick. But I would like to see '--optimal' argument tested as well.

tbowers7 committed:
Add test for `--optimal` to `test_compare_sky`

	modified:   pypeit/tests/test_scripts.py

tbowers7 commented:
On branch `develop` against dev suite `develop` branch and Cooked 1.4.5:
(This is the same failure on my machine as happened in #1216 -- source unknown.)
```
=========================== short test summary info ============================
FAILED pypeit/tests/test_setups.py::test_setup_vlt_fors2 - TypeError: NoneTyp...
========== 1 failed, 287 passed, 4500 warnings in 1597.22s (0:26:37) ===========
_____________________________ test_setup_vlt_fors2 _____________________________

    @dev_suite_required
    def test_setup_vlt_fors2():
...
        # Now chk calib
        pargs = ChkForCalibs.parse_args([droot, '-s', 'vlt_fors2'])
>       answers, ps = ChkForCalibs.main(pargs)

/home/tbowers/d1/codes/PypeIt/pypeit/tests/test_setups.py:536:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
args = Namespace(root='/home/tbowers/d1/codes/PypeIt-development-suite/RAW_DATA/vlt_fors2/300I//FORS2', spectrograph='vlt_fors2', extension='.fits', save_setups=False)

    @staticmethod
    def main(args):
...
        # Setup the table. Need to use object type for strings so that
        # they're not truncated.
        answers = table.Table()
        answers['setups'] = list(uniq_cfg.keys())
        # Add the configuration columns
        for setup, setup_dict in uniq_cfg.items():
            for key, value in setup_dict.items():
                answers[key] = np.empty(len(answers), dtype=object) if isinstance(value, str) \
>                                   else type(value)(0)
E               TypeError: NoneType takes no arguments

/home/tbowers/d1/codes/PypeIt/pypeit/scripts/chk_for_calibs.py:90: TypeError
```

On branch `fix_script_compare_sky` against dev suite `develop` branch and Cooked 1.4.5:
(Identical failure to branch `develop`.)
```
=========================== short test summary info ============================
FAILED pypeit/tests/test_setups.py::test_setup_vlt_fors2 - TypeError: NoneTyp...
========== 1 failed, 288 passed, 4500 warnings in 1520.29s (0:25:20) ===========
```
Note the additional passing test on the `fix_script_compare_sky` branch is the test added in this PR.
